scripts/encode2ga4gh.py

```diff
@@ -0,0 +1,296 @@
+                writeSamstats(samOutfile, samstats, analysisId)
+
+
+def write_dist_tables(data):
+    log("Writing distribution tables")
+    for analysisId, distfile in data:
+        # output table
+        distTable = os.path.join(analysisId, "dist.table")
+
+        # write mapping distribution table
+        distribution = getDistribution(distfile)
+        with open(distTable, "w") as distOutfile:
+            writeDistribution(distOutfile, distribution, analysisId,
+                              distribution["mapped"])
+
+
+def write_expression_tables(data, annotationId, output_folder):
+    log("Writing gene expression tables")
+    for analysisId, quantfile in data:
+        # output table
+        expTable = os.path.join(output_folder, analysisId, "expression.table")
+
+        # write expression table
+        print("processing {}".format(analysisId))
+        with open(expTable, "w") as quantOutfile:
+            write_expression(analysisId, annotationId, quantfile, quantOutfile)
+
+
+def makeParser(usage):
+    """
+        Default values will download and parse 4 quantifications from the
+        ENCODE Evaluation dataset as an example set of test data.
+    """
+
+    parser = optparse.OptionParser(usage=usage)
+    parser.add_option("--dataset", dest="dataset")
+    parser.add_option("--description", dest="description")
+    parser.add_option("--annotation", dest="annotationId")
+    parser.add_option("--file_limit", type="int", dest="subset")
+
+    parser.set_defaults(dataset="ENCSR000AJW", annotationId="Gencodev16",
+                        description="RNAseq data from ENCODE evaluation",
+                        subset=4)
+
+    return parser
+
+
+def main(argv):
+
+    usage = "Usage: {} <data-folder>".format(argv[0])
+    if len(argv) != 2:
+        print(Color.red(usage))
+        sys.exit(1)
+
+    parser = makeParser(usage)
+    (options, args) = parser.parse_args(argv[1:])
+
+    host = "https://www.encodeproject.org"
+    dataset = options.dataset
+    url = "{host}/datasets/{dataset}/?frame=embedded".format(host=host,
+                                                             dataset=dataset)
+    headers = {
+        'Accept': 'application/json; charset=utf-8'
+    }
+
+    data_type = "gene quantifications"
+    data_folder = argv[1]
+    rna_folder = "rnaQuant"
+    output_folder = os.path.join(data_folder, rna_folder)
+    subset = options.subset
+
+    log("Downloading GA4GH test dataset - RNA Quantification API")
+    print("ENCODE dataset: {}".format(Color.blue(dataset)))
+    print("data type:      {}".format(Color.blue(data_type)))
+    print("subset size:    {}".format(Color.blue(subset)))
+    print("output folder:  {}".format(Color.blue(output_folder)))
+    get_data_from_host(url, headers, host, data_type, output_folder,
+                       options.description, options.annotationId, subset)
+
+    log("DONE")
+
+
+if __name__ == '__main__':
+    main(sys.argv)
```

Comment:
a lot of snake_case crept into this file... please change all identifiers to camelCase